README.md

```diff
@@ -0,0 +1,42 @@
+# Operator Custom Metrics
+This library assists in collecting operator custom-metrics from operators and exposing them to a prometheus instance. The user can define their own set of metrics for a specific operator, and the library would help in integrating these custom metrics with a prometheus instance for monitoring purposes.
+
+## Description
+This library is designed to provide the user with an easy procedure to register custom metrics with a running prometheus instance. The user can provide the **metrics configuration** based on which the metrics are collected. The library currently supports openshift-operators and hence, to access the metrics routes are created. 
+The library simplifies the following processes:
+1. Registering: Register custom metrics provided by the user to prometheus.
+2. Publishing: Publish metrics at a specified endpoint and path.
+3. Creating resources:
+    1. Create a service resource. 
+    2. Create an external [route](https://docs.openshift.com/container-platform/3.9/architecture/networking/routes.html) so that the metrics can be accessed. 
+
+The following are the parameters of the metrics configuration (`metricsConfig`) for the custom metrics:
+
+- Metrics Path:
+This is the endpoint at which the metrics would be exposed. The default metrics endpoint is `/customMetrics`.
+
+- Metrics Port:
+This is the port at which the metrics would be published. The default metrics port is `:8089`
+
+- List of collectors:
+The user can provide the collectors which are to be registered with prometheus. If no collectors are passed, the default GoLang metrics are exposed.
```

Comment:
The default metrics are exposed either way, and they should referenced as `default prometheus server metrics`